src/App.tsx

```diff
@@ -1,37 +1,96 @@
-import React from 'react';
-import Header from './components/Header/Header';
+import './App.scss'
+
+import React, {useMemo, createContext, useEffect} from 'react';
 import TodoList from './components/TodoList/TodoList';
 import AddTodo from './components/AddTodo/AddTodo';
-import { Box } from '@mui/material';
-import { makeStyles, createStyles } from '@mui/styles';
+import {Box, IconButton, Card, CardHeader, CardContent, CardActions} from '@mui/material';
+import {ThemeProvider, createTheme} from '@mui/material/styles';
+import Brightness4Icon from '@mui/icons-material/Brightness4';
+import Brightness7Icon from '@mui/icons-material/Brightness7';
+import useMediaQuery from '@mui/material/useMediaQuery';
 
-const useStyles = makeStyles(() =>
-  createStyles({
-    appWrapper: {
-      backgroundColor: '#f8f9fa',
-      height: '100vh'
-    },
-    todoListWrapper: {
-      margin: '0 auto',
-      border: '1px solid black',
-      borderRadius: '5px',
-      minHeight: '50vh',
-      width: '100%',
-      maxWidth: 360,
-    }
-  })
-);
+import {switchTheme, saveState, LIGHT, DARK} from './redux/appSlice';
+import {useDispatch, useSelector} from 'react-redux';
+import {RootState} from './redux/store';
+
+const ColorModeContext = createContext({
+  toggleColorMode: (mode: 'light' | 'dark' | undefined) => {
+  }
+});
 
 const App = () => {
-  const classes = useStyles();
+  const dispatch = useDispatch();
+  const mode = useSelector((state: RootState) => state.config.mode);
+  const prefersColorScheme = useMediaQuery('(prefers-color-scheme: dark)') ? DARK : LIGHT;
+  useEffect(() => {
+    if (!mode) {
+      dispatch(switchTheme(prefersColorScheme));
+    }
+  }, []);
+  useEffect(() => {
+    if (mode) {
+      saveState(mode);
+    }
+  }, [mode]);
+  const colorMode = useMemo(
+    () => ({
+      toggleColorMode: (mode: 'light' | 'dark' | undefined) => {
+        dispatch(switchTheme(mode === LIGHT ? DARK : LIGHT));
+      },
+    }),
+    [],
+  );
```

Comment:
```suggestion
  const mode = useSelector((state: RootState) => state.config.mode);
  
  const dispatch = useDispatch();
  
  const prefersColorScheme = useMediaQuery('(prefers-color-scheme: dark)') ? DARK : LIGHT;
  
  useEffect(() => {
    if (!mode) {
      dispatch(switchTheme(prefersColorScheme));
    }
  }, []);
  
  useEffect(() => {
    if (mode) {
      saveState(mode);
    }
  }, [mode]);
  
  const colorMode = useMemo(
    () => ({
      toggleColorMode: (mode: 'light' | 'dark' | undefined) => {
        dispatch(switchTheme(mode === LIGHT ? DARK : LIGHT));
      },
    }),
    [],
  );
```

It's always better with logical new lines :)

Reply:
I'm not sure what I should fix in this point... Could you please get me a little bit more information?

Reply:
Add logical spaces between lines

Reply:
Fixed